src/basic/process-util.c

```diff
@@ -1598,12 +1599,8 @@ int safe_fork_full(
                 }
         }
 
-        if ((flags & (FORK_NEW_MOUNTNS|FORK_NEW_USERNS|FORK_NEW_NETNS|FORK_NEW_PIDNS)) != 0)
-                pid = raw_clone(SIGCHLD|
-                                (FLAGS_SET(flags, FORK_NEW_MOUNTNS) ? CLONE_NEWNS : 0) |
-                                (FLAGS_SET(flags, FORK_NEW_USERNS) ? CLONE_NEWUSER : 0) |
-                                (FLAGS_SET(flags, FORK_NEW_NETNS) ? CLONE_NEWNET : 0) |
-                                (FLAGS_SET(flags, FORK_NEW_PIDNS) ? CLONE_NEWPID : 0));
+        if (FLAGS_SET(flags, FORK_NEW_PIDNS))
+                pid = raw_clone(SIGCHLD|CLONE_NEWPID);
```

Comment:
hmm this doesn't look right. we want that the the pidns is owned by the userns if both are created together.

jence, i am pretty sure you should create them altogether here as before if PIDNS is on.

(not all ns types are alike: user ns own other ns types)

Reply:
then add a comment about that pls